### DISCUSSION TITLE
关于后期实现自动验证的思路

### DISCUSSION THREAD

**Discussion by Amorter** [mikumifa/biliTickerBuy#84] [提问专区]
目前打算写一个爬虫爬取b站选字验证图片，训练自己的孪生神经网络模型(对比两张图片的相似度)，依次将题目分割后的图片与背景图经过目标检测分割的图片进行比对，不知道各位有没有更好的想法(或者更容易的方法也可以，这个实现方法太麻烦)

**Comment by mikumifa** [comment]
- 要做成外置的，独立于程序，原因
  - 防止gradio 逻辑太复杂，不好维护
  - 想要降低门槛，不想安装太复杂， 现在装个驱动能难倒一堆人，装个node不是要命
  - 可以通过请求方式，调用外置的验证码处理程序
  - 或者 直接外置点击
- 正确性肯定是没办法接近100%的， 所以就要给出完善的补救措施
  - 能够重试， 这一点非常重要

**Reply by Amorter** [reply]
> * 要做成外置的，独立于程序，原因
>   
>   * 防止gradio 逻辑太复杂，不好维护
>   * 想要降低门槛，不想安装太复杂， 现在装个驱动能难倒一堆人，装个node不是要命
>   * 可以通过请求方式，调用外置的验证码处理程序
>   * 或者 直接外置点击
> * 正确性肯定是没办法接近100%的， 所以就要给出完善的补救措施
>   
>   * 能够重试， 这一点非常重要

请求方式有两种实现可能
- gt验证程序部署在本地，开启http服务器，调用相关逻辑。（这种方式我朋友做的软件使用过，由于用户电脑上面未知的防火墙和杀毒软件限制，经常出bug，甚至操作系统的字符集也会导致软件bug）
- 服务器部署在远程 （服务器成本问题？法律风险？）

最好的选择还是尝试通过python直接io调用本地程序，可以将gt验证程序打包，然后使用本项目调用

重试逻辑也有两种实现可能
- 本项目调用gt验证模块，模块返回success或fail，本项目收到fail后主动重试。
- 本项目调用gt验证模块，模块fail后自动重试，直到success。

构想是通过传一个bool的retry参数，同时支持两种功能

目标的点选验证码识别准确率为60%以上，我将会持续跟进开发

**Comment by mikumifa** [comment]
通信都是小事，感觉破极验难度极高，用yolo的话速度特别慢，我觉得能用ocr识图，然后能够不停的重试就挺好的

**Reply by Amorter** [reply]
好的，那其实现在这个demo差不多满足需求了，我正在把这些东西封装成一个模块，到时候这个项目通过子进程或者FFI调用就行了

**Comment by WittF** [comment]
有考虑过接入现成的反验证码服务吗, 我自己用2captchaAPI浏览器插件在训练里试了3次, 平均每次需要15秒解决, 但需要付费, 1次0.003美元差不多100次两块三毛

**Reply by WittF** [reply]
capsolver的按次购会便宜不少, 
![Capsolver-Geetest](https://cr.wittf.site/f/AQHE/Capsolver-Geetest.jpg)

**Reply by mikumifa** [reply]
> capsolver的按次购会便宜不少, 
> ![Capsolver-Geetest](https://cr.wittf.site/f/AQHE/Capsolver-Geetest.jpg)
> 

这不错
